File: Leetcode/2563. Count the Number of Fair Pairs.cpp
class Solution {
public:
    long long countFairPairs(vector<int>& nums, int lower, int upper) {
        sort(nums.begin(),nums.end());
        int n=nums.size();
        int res=0;
        for(int i=0;i<n;i++){
            int min = lower - nums[i] ,max= upper - nums[i];
            auto high=upper_bound(nums.begin()+i+1,nums.end(),max);
            auto low=lower_bound(nums.begin()+i+1,nums.end(),min);
            res+=(high-low);
        }
        return res;
    }
};

File: Leetcode/2563. Count the Number of Fair Pairs.py
class Solution:
    def countFairPairs(self, nums: List[int], lower: int, upper: int) -> int:
        nums.sort()
        res=0
        n=len(nums)
        if n==1:
            return 0
        for i in range(n):
            minimum = lower-nums[i]
            maximum = upper-nums[i]
            low = bisect_left(nums, minimum, i + 1)
            high = bisect_right(nums, maximum, i + 1)
            res += high - low
        return res
        
